gnmi_server/server_test.go

```diff
@@ -2742,6 +2743,100 @@ func TestAuthCapabilities(t *testing.T) {
 	}
 }
 
+func createCPUStat(increment uint64) (*linuxproc.Stat) {
+    var stat linuxproc.Stat = linuxproc.Stat{}
+    cpuStat := linuxproc.CPUStat{}
+    cpuStat.User = increment
+    cpuStat.Nice = increment
+    cpuStat.System = increment
+    cpuStat.IRQ = increment
+    cpuStat.SoftIRQ = increment
+    stat.CPUStatAll = cpuStat
+    return &stat
+}
+
+func TestCPUUtilization(t *testing.T) {
+    // will mock linuxproc.ReadStat
+    var increment uint64 = 0
+    mock := gomonkey.ApplyFunc(linuxproc.ReadStat, func(path string) (*linuxproc.Stat, error) {
+        t.Log("mock linuxproc.ReadStat is called")
+	t.Log(increment)
+        stat := createCPUStat(increment)
+	t.Log(stat)
+        increment += 1
+        return stat, nil
+    })
+
+    defer mock.Reset()
+    s := createAuthServer(t, 8081)
+    go runServer(t, s)
+    defer s.s.Stop()
+
+    /*q := createQueryOrFail(t,
+        pb.SubscriptionList_POLL,
+        "OTHERS",
+        []subscriptionQuery{
+            {
+                Query:    []string{"platform", "cpu"},
+                SubMode:  pb.SubscriptionMode_SAMPLE,
+                SampleInterval: uint64(10*time.Second),
+            },
+        },
+        false)
+
+    q.Addrs = []string{"127.0.0.1:8081"}
+    */
+    tests := []struct {
+        desc        string
+	q           client.Query
+	wantNoti    []client.Notification
+	poll        int
+    }{
+        {
+            desc: "poll query for CPU Utilization",
+	    poll: 10,
+	    q: client.Query{
+                Target: "OTHERS",
+		Type:    client.Poll,
+		Queries: []client.Path{{"platform/cpu"}},
+		TLS:     &tls.Config{InsecureSkipVerify: true},
+	    },
+	    wantNoti: []client.Notification{
+                client.Connected{},
+		client.Sync{},
+	    },
+        },
+    }
+
+    for _, tt := range tests {
+        t.Run(tt.desc, func(t *testing.T) {
+            q := tt.q
+	    q.Addrs = []string{"127.0.0.1:8081"}
+            c := client.New()
+            defer c.Close()
+            var gotNoti []string
+            q.NotificationHandler = func(n client.Notification) error {
+                t.Log("Inside of notification handler")
+                if nn, ok := n.(client.Update); ok {
+                    nn.TS = time.Unix(0, 200)
+                    str := fmt.Sprintf("%v", nn.Val)
+		    t.Log(str)
+                    gotNoti = append(gotNoti, str)
+                }
+                return nil
+            }
+
+            go func() {
+                c.Subscribe(context.Background(), q)
+            }()
+
+            t.Log(len(gotNoti))
+	    // t.Log(gotNoti[0])
+        })
+    }
+    s.s.Stop()
```

Comment:
This is not needed. Code in line 2773 will call `s.s.Stop()`

Reply:
Removed